### PR TITLE
(StaticHtml|Excel)Reporter: Use `labels` instead of `data` for the metadata section

### PR DESCRIPTION
**Pull request by fviernau**
The changes prepare for the removal of `OrtResult.data`. 
Please see the individual commit messages.

### CODE REVIEW

File: reporter/src/main/kotlin/utils/ReportTableModelMapper.kt [hunk 1]
```diff
@@ -227,13 +227,7 @@ class ReportTableModelMapper(
...
         )
 
-        val metadata = mutableMapOf<String, String>()
-        (ortResult.data["job_parameters"] as? Map<*, *>)?.let {
-            it.entries.associateTo(metadata) { (key, value) -> key.toString() to value.toString() }
-        }
-        (ortResult.data["process_parameters"] as? Map<*, *>)?.let {
-            it.entries.associateTo(metadata) { (key, value) -> key.toString() to value.toString() }
-        }
+        val metadata = ortResult.labels.mapKeys { it.key.substringAfter(".") }
```
Comment (sschuberth) [inline reporter/src/main/kotlin/utils/ReportTableModelMapper.kt:230]: I looks a bit arbitrary to swallow the part before the first `.`, and assuming that `.` as part of a label has a special meaning is not really generic. While I'd be fine to introduce the latter as a convention, I believe we should not omit parts before `.` but use them for some visual grouping as part of the reports.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: reporter/src/main/kotlin/utils/ReportTableModelMapper.kt (hunk 1 above)
Comment (sschuberth) [reply reporter/src/main/kotlin/utils/ReportTableModelMapper.kt:230]: Is what I wriote above what you mean by

> Note: A reasonable improvement would be to introduce sections for labels with the same prefix.

in the commit message?

Comment (fviernau) [reply reporter/src/main/kotlin/utils/ReportTableModelMapper.kt:230]: yes, exactly. Would you be fine doing this in a future PR.

Comment (sschuberth) [reply reporter/src/main/kotlin/utils/ReportTableModelMapper.kt:230]: Ok, but could you then please, in addition to the commit message, add an according `TODO` for this line?

Also, shouldn't this commit then remove the changes done to `reporter/src/funTest/assets/static-html-reporter-test-input.yml` in the first commit?

Comment (fviernau) [reply reporter/src/main/kotlin/utils/ReportTableModelMapper.kt:230]: > Also, shouldn't this commit then remove the changes done to reporter/src/funTest/assets/static-html-reporter-test-input.yml in the first commit?

There will be a different PR soon which removes the custom data. Imo it'd better fit in there.

Comment (fviernau) [reply reporter/src/main/kotlin/utils/ReportTableModelMapper.kt:232]: I added a TODO comment.

**Review by sschuberth — DISMISSED** [review]

**Review by sschuberth — APPROVED** [review]

File: reporter/src/main/kotlin/utils/ReportTableModelMapper.kt (hunk 1 above)
Comment (fviernau) [reply reporter/src/main/kotlin/utils/ReportTableModelMapper.kt:232]: I've turned you proposal above into an issue, see #3062 3062.

**Pull request merged**